src/Sylius/Bundle/CartBundle/Controller/CartItemController.php

```diff
@@ -137,4 +135,47 @@ public function removeAction(Request $request)
 
         return $this->redirectToCartSummary($configuration);
     }
+
+    /**
+     * @param string $flashMessage
+     * @param MetadataInterface $metadata
+     *
+     * @return string
+     */
+    private function translateMessage($flashMessage, MetadataInterface $metadata)
+    {
+        return $this->get('translator')->trans($flashMessage, ['%resource%' => ucfirst($metadata->getHumanizedName())], 'flashes');
+    }
+
+    /**
+     * @param CartInterface $cart
+     */
+    private function recalculateCart(CartInterface $cart)
+    {
+        $orderRecalculator = $this->get('sylius.order_processing.order_recalculator');
+        $orderRecalculator->recalculate($cart);
+
+        $cartManager = $this->get('sylius.manager.cart');
+        $cartManager->persist($cart);
+        $cartManager->flush();
+    }
+
+    /**
+     * @param CartInterface $cart
+     * @param CartItemInterface $item
+     */
+    private function resolveCartItem(CartInterface $cart, CartItemInterface $item)
```

Comment:
This logic can live here for now, but later we will think about moving it somewhere else. :)